lib/sign_in/idme/service.rb

```diff
@@ -147,18 +165,18 @@ def jwe_decrypt(encrypted_jwe)
       end
 
       def jwt_decode(encoded_jwt)
-        with_validation = true
+        verify_expiration = true
         decoded_jwt = JWT.decode(
           encoded_jwt,
-          config.jwt_decode_public_key,
-          with_validation,
-          {
-            verify_expiration: with_validation,
-            algorithm: config.jwt_decode_algorithm
-          }
+          nil,
+          verify_expiration,
+          { verify_expiration:, algorithm: config.jwt_decode_algorithm, jwks: get_public_jwks }
         )&.first
```

Comment:
❓ I know you didn't touch this, but the use of the safe navigator here makes me wonder if it makes sense to handle a `nil` decoded JWT differently here. Maybe raise an `JWT::DecodeError` if the decoded JWT is `nil`?

```ruby
decoded_jwt = JWT.decode(
  encoded_jwt,
  nil,
  verify_expiration,
  { verify_expiration, algorithm: config.jwt_decode_algorithm, jwks: get_public_jwks }
)
raise JWT::DecodeError if decoded_jwt.blank?
```

Reply:
[It doesn't look](https://github.com/jwt/ruby-jwt/blob/main/lib/jwt/decode.rb#L24) like `JWT.decode` can return `nil`, so I wonder why the safe navigation operator was introduced here. We could probably just call `.first` without using safe navigation, but if there is a chance that the decoded token could be `nil`, we should probably handle it.

Reply:
I think the error handling should cover any instances that would result in a `nil` jwt so I'll remove the safe navigator in these